app/src/app/[lng]/[inventory]/data/[step]/[subsector]/page.tsx

```diff
@@ -129,14 +134,17 @@ function SubSectorPage({
   };
 
   const getFilteredSubsectorScopes = () => {
+    if (!inventoryData) return [];
     return Object.entries(MANUAL_INPUT_HIERARCHY)
       .filter(([key]) => key.startsWith(subSectorData?.referenceNumber!))
-      .map(([k, v]) => ({
-        ...v,
-        referenceNumber: k,
-      }));
+      .map(([k, v]) => ({ ...v, referenceNumber: k }))
+      .filter((scope) =>
+        getScopesForInventoryAndSector(
+          inventoryData.inventoryType!,
+          scope.referenceNumber[0],
+        )!.includes(scope.scope),
```

Comment:
Should we handle this null error instead of ignoring it with `!`? Just a default value of [] would be sufficient here (or throwing an error in case that's better).